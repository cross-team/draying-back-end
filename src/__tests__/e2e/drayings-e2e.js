// import our production apollo-server instance
import { server } from '../../'
import gql from 'graphql-tag'
import { startTestServer, toPromise } from '../__utils'

// const DRAYING_LIST_QUERY = gql`
//   query GetDrayings {
//     drayings(first: 25) {
//       __typename
//       totalCount
//       pageInfo {
//         startCursor
//         endCursor
//         hasNextPage
//         hasPreviousPage
//       }
//       edges {
//         node {
//           id
//           booking
//           client {
//             id
//           }
//           containerSize {
//             name
//           }
//           container
//           __typename
//         }
//       }
//     }
//   }
// `

const LAUNCH_LIST_QUERY = gql`
  {
    drayings(first: 25) {
      __typename
      totalCount
      pageInfo {
        startCursor
        endCursor
        hasNextPage
        hasPreviousPage
      }
      edges {
        node {
          id
          booking
          client {
            id
          }
          containerSize {
            name
          }
          container
          __typename
        }
      }
    }
  }
`

describe('Server - e2e', () => {
  let stop, graphql

  beforeEach(async () => {
    const testServer = await startTestServer(server)
    stop = testServer.stop
    graphql = testServer.graphql
  })

  afterEach(() => stop())

  it('gets list of drayings`', async () => {
    try {
      const res = await toPromise(
        graphql({
          query: LAUNCH_LIST_QUERY,
        }),
      )
      expect(res).toMatchSnapshot()
    } catch (error) {
      console.log(`Error Promise: ${error}`)
    }
  })
})
