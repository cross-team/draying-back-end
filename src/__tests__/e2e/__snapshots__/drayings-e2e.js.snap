// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server - e2e gets list of drayings\` 1`] = `
Object {
  "data": null,
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "INTERNAL_SERVER_ERROR",
      },
      "locations": Array [
        Object {
          "column": 3,
          "line": 2,
        },
      ],
      "message": "Cannot read property '.DRAYINGAUTH' of undefined",
      "path": Array [
        "drayings",
      ],
    },
  ],
}
`;
